### PR TITLE
Substrate governor update 

### LINKED ISSUE #367: [TASK] Add Tests for Substrate governor updates flow
## Overview
Add governor update test case for substrate based chains. Relayer should watch for `PublicKeySignatureChanged` and relay `set_maintainer` transaction to substrate chain.

```rust
// msg = nonce+dkg_key
 let mut message = nonce.to_be_bytes().to_vec();
 message.extend_from_slice(&new_maintainer);

 let set_maintainer_call = SetMaintainer {
            message: BoundedVec(message.clone()),
            signature: BoundedVec(signature.clone()),
 };

```

----
### Checklist
- [ ] Update relayer code to construct `msg = nonce+dkg_key`
- [ ] Write governor update test

### PR DESCRIPTION
**Pull request by salman01zp**
## Summary of changes
- [x] Send `nonce+dkg key` to signature bridge to update maintainer
- [x] Governor update flow test for substrate

### Reference issue to close (if applicable)
- Closes #367 

- Depends on https://github.com/webb-tools/protocol-substrate/pull/305
- Depends on https://github.com/webb-tools/webb-rs/pull/125
- Depends on https://github.com/webb-tools/protocol-substrate/pull/306
- Depends on https://github.com/webb-tools/webb.js/pull/310
-----
### Code Checklist 

- [ ] Tested
- [ ] Documented

### CODE REVIEW

**Comment by salman01zp** [comment]
We will need to force set maintainer nonce too along with maintainer/governor. I have drafted a PR for it [here](https://github.com/webb-tools/protocol-substrate/pull/305).

**Comment by salman01zp** [comment]
will need to update `protocol-substrate-types` and rust bindings in `webb-rs`.

**Comment by shekohex** [comment]
> We will need to force set maintainer nonce too along with maintainer/governor. I have drafted a PR for it [here](https://github.com/webb-tools/protocol-substrate/pull/305).

This PR is now merged!

**Comment by shekohex** [comment]
44

> will need to update `protocol-substrate-types` and rust bindings in `webb-rs`.

the PR is up here https://github.com/webb-tools/webb-rs/pull/125

File: tests/lib/substrateNodeBase.ts [hunk 1]
```diff
@@ -47,334 +36,186 @@ export type UsageMode = DockerMode | HostMode;
...
               const decoded = api.registry.findMetaError(
                 dispatchError.asModule
               );
               const { docs, name, section } = decoded;
-              reject(`${section}.${name}: ${docs.join(' ')}`);
+
+              reject(new Error(`${section}.${name}: ${docs.join(' ')}`));
             } else {
-              // Other, CannotLookup, BadOrigin, no extra info
+            // Other, CannotLookup, BadOrigin, no extra info
```
Comment (drewstone) [inline tests/lib/substrateNodeBase.ts:183]: Spacing off?

File: tests/lib/substrateNodeBase.ts [hunk 2]
```diff
@@ -47,334 +36,186 @@ export type UsageMode = DockerMode | HostMode;
...
-        ({ status, dispatchError }) => {
-          // status would still be set, but in the case of error we can shortcut
-          // to just check it (so an error would indicate InBlock or Finalized)
+        ({ dispatchError, status }) => {
+        // status would still be set, but in the case of error we can shortcut
+        // to just check it (so an error would indicate InBlock or Finalized)
           if (dispatchError) {
             if (dispatchError.isModule) {
-              // for module errors, we have the section indexed, lookup
+            // for module errors, we have the section indexed, lookup
```
Comment (drewstone) [inline tests/lib/substrateNodeBase.ts:175]: Spacing off?

**Review by drewstone — COMMENTED** [review]
Looks good, double check spacing,need more reviews but looks good otherwise.

**Review by shekohex — COMMENTED** [review]
Looks good, try to run `yarn format` inside the tests to format all typescript files.

**Review by shekohex — APPROVED** [review]

File: crates/event-watcher-traits/src/evm/bridge_watcher.rs [hunk 3]
```diff
@@ -62,7 +62,9 @@ where
                 .map_err(Into::into)
                 .map_err(backoff::Error::transient)
                 .await?;
-            let bridge_key = BridgeKey::new(my_chain_id);
+            let typed_chain_id =
```
Comment (drewstone) [inline crates/event-watcher-traits/src/evm/bridge_watcher.rs:65]: Can the other variable be deleted?

**Review by drewstone — CHANGES_REQUESTED** [review]

**Comment by Nutomic** [comment]
It seems that this change broke integration tests, this is what I get on develop:
```
yarn && yarn test

TSError: ⨯ Unable to compile TypeScript:
test/evm/evmToSubstrateTransaction.test.ts:298:78 - error TS2554: Expected 7 arguments, but got 8.

298     await vanchor1.transact([], [depositUtxo], 0, 0, '0', '0', tokenAddress, {
~
299       [typedSourceChainId]: leaves,
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
300     });
~~~~~
test/evm/evmToSubstrateTransaction.test.ts:331:7 - error TS2345: Argument of type 'import("/home/felix/workspace/webb/relayer/tests/node_modules/@webb-tools/test-utils/node_modules/@webb-tools/anchors/lib/VAnchor").VAnchor' is not assignable to parameter of type 'import("/home/felix/workspace/webb/relayer/tests/node_modules/@webb-tools/anchors/lib/VAnchor").VAnchor'.
Type 'VAnchor' is missing the following properties from type 'VAnchor': leafCreatedAtBlock, updateTreeOrForestState, generateProof, register, and 14 more.

331       vanchor1,
~~~~~~~~

at createTSError (/home/felix/workspace/webb/relayer/tests/node_modules/ts-node/src/index.ts:859:12)
at reportTSError (/home/felix/workspace/webb/relayer/tests/node_modules/ts-node/src/index.ts:863:19)
at getOutput (/home/felix/workspace/webb/relayer/tests/node_modules/ts-node/src/index.ts:1077:36)
at Object.compile (/home/felix/workspace/webb/relayer/tests/node_modules/ts-node/src/index.ts:1433:41)
at transformSource (/home/felix/workspace/webb/relayer/tests/node_modules/ts-node/src/esm.ts:400:37)
at /home/felix/workspace/webb/relayer/tests/node_modules/ts-node/src/esm.ts:278:53
at async addShortCircuitFlag (/home/felix/workspace/webb/relayer/tests/node_modules/ts-node/src/esm.ts:409:15)
at async nextLoad (node:internal/modules/esm/loader:163:22)
at async ESMLoader.load (node:internal/modules/esm/loader:605:20)
at async ESMLoader.moduleProvider (node:internal/modules/esm/loader:457:11)
at async link (node:internal/modules/esm/module_job:67:21)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by salman01zp** [comment]
did you try yarn install ?

**Comment by Nutomic** [comment]
Just tried it, same errors as before.

**Comment by salman01zp** [comment]
Clearning node modules, cache, and reinstalling should fix it.

**Pull request merged**